### ISSUE TITLE
Not visible in VS 20015 Pro

### ISSUE THREAD

**Issue by RajNZ** [oleksabor/GroupMembershipPolicy#1]
I compiled in VS 2015 and installed the vsix, but it doesn't appear in the list of available check-in policies in VS 2015 Pro.
I then changed the project reference for Microsoft.TeamFoundation.VersionControl.Client (and some other references also) to version 15, the version that matches VS 2015, and recompiled and reinstalled but again it doesn't appear in the list of available check-in policies in VS 2015 Pro.

Do you have any ideas why I am having this problem?
Thanks.

**Comment by oleksabor** [comment]
hi
Sorry that this does not work. I've heard about success installation of this policy on another machines. Exactly with VS Pro 2015. 
Can you please check with some other policy (like ColinsALMCheckinPolicies) is it visible or not?

I've googled a bit. Please check registry path (HKEY_CURRENT_USER\Software\Microsoft\VisualStudio\11.0_Config\TeamFoundation\SourceControl\Checkin Policies) on your computer. Is there CustomGroupMembershipPolicy value with path installed? (http://stackoverflow.com/questions/14571063/tfs-custom-check-in-policy-is-not-in-the-add-box)

Do you have this extension (CustomGroupMembershipPolicy) enabled ?

What TFS server version do you have? I've tried with tfs 2015 afair.

Regards

**Comment by RajNZ** [comment]
Hi,

I'm using VS 2015 Pro 32 bit and TFS 2015, running in Win 8 64 bit.
ColinsALMCheckinPolicies is visible and is working and so is the Custom Path policy.
CustomGroupMembershipPolicyInstaller appears under "Extensions and Updates" and is enabled, but CustomGroupMembershipPolicy does not appear under "Add Check-in Policy".

Curiously, if I compile the dll as 64 bit and reinstall then it does appear in the list, but as something in an incorrect format that cannot be used.  Fair enough, to be expected.
At least that shows that the installation process is working.

The registry is correct : HKEY_CURRENT_USER\Software\Microsoft\VisualStudio\14.0_Config_9460\TeamFoundation\SourceControl\Checkin Policies.
- Note that every time I uninstall and reinstall the vsix a new registry key is created for VS - I started with 14.0_Config and now I am up to 14.0_Config_9460!  Basically, wherever ColinsALMCheckinPolicies  is mentioned, CustomGroupMembershipPolicy is mentioned also.
CustomGroupMembershipPolicy.dll exists in the directory that the registry key refers to, along with various supporting assemblies.

I've read a few articles on this problem and as far as I can see everything is in the right place and should work!  So I'm baffled at the moment.
I was hoping you knew some trick during compilation to make it visible to VS.

Thanks anyway.

**Comment by oleksabor** [comment]
hi

wait a second please
VS is running as 32bit application for sure (I have win 8.1 x64)
so to make it possible to run policy assembly on tfs service side and on the machine I'm building policy as Debug Any CPU (it should be release I suppose). So assembly is running as x86 if it is loaded in VS
and as x64 if it is loaded with x64 process (TFS service I suppose). Settings are serialized|deserialized when configuration dialog loads policy configuration data from TFS and writes it back.
What exactly build have you tried? Could you please try with Debug|AnyCPU one more time ?

Also it is strange to hear about "14.0_Config_9460". Is it key registry name?
I have exactly HKEY_CURRENT_USER\Software\Microsoft\VisualStudio\14.0_Config\TeamFoundation\SourceControl\Checkin Policies registry path. There is no additional 14.0_Config_something keys.

Please try with Debug|AnyCPU and let me know

Regards

**Comment by RajNZ** [comment]
Hi,

I have built it as Debug/AnyCPU, but still have the problem.

Yeah, the registry key thing is strange.  It seems that VS makes temporary registry keys when changes are made and it eventually removes them and rolls the changes back into 14.0_Config.  I saw the extra numbered keys (there were a few) deleted when I closed VS one day and so now I only have 14.0_Config.
Weird.
Anyway, the registry value looks ok and the files exist in the right place, so not sure what else to try.
Thanks for your help.

**Comment by oleksabor** [comment]
hi

I've just checked versions numbers
So I have 	Windows 8.1 x64
Microsoft Visual Studio Professional 2015 Version 14.0.25431.01 Update 3
and I've just installed locally TFS 2015 14.0.23129.1

groupmembershippolicy is visible in the VS2015 add dialog when I has connected to the local tfs instance. And it appears in the TeamProject\SourceControl\Checkinpolicy list

Do you restart VS after policy installation?
![clipboard115](https://cloud.githubusercontent.com/assets/16188507/25269633/34fd3a26-2686-11e7-9f5d-3556c2b58b81.jpg)

Regards

ps: please note that I'm testing v1.8 release that is built as Debug|AnyCpu configuration. It contains changed referenced tfs assembly versions to 14.0 (but with <SpecificVersion>False</SpecificVersion>) but generally is the same as 1.7

**Comment by RajNZ** [comment]
I see the problem now.
I had changed the references for some TeamFoundation assemblies to version 15 and some to version 14 - they need to all be version 14 for VS 2015.
So, I started again with a fresh copy of your solution and then installed v14.102 of the Microsoft.TeamFoundationServer.Client and Microsoft.TeamFoundationServer.ExtendedClient NuGet packages; after compiling and reinstalling the policy appears in the list.  Yay!!!

Thank you for your help with this, I appreciate it.
